dbstore.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/bmatsuo/lmdb-go/lmdb"
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+const (
+	DBStoreName = "mender-store"
+)
+
+// DBStore is an opaque structure representing a database backed storage.
+// Implements `Store` interface.
+type DBStore struct {
+	env *lmdb.Env
+}
+
+type DBStoreWrite struct {
+	io.WriteCloser
+	dbs  *DBStore
+	name string
+	data bytes.Buffer
+}
+
+// NewDBStore creates an instance of Store backed by LMDB database. DBStore uses
+// a single file for DB data (named `DBStoreName`). Parameter `dirpath` is a
+// directory where the file will be stored. Returns nil if initialization
+// failed.
+func NewDBStore(dirpath string) *DBStore {
+	env, err := lmdb.NewEnv()
+	if err != nil {
+		log.Errorf("failed to create DB environment: %v", err)
+		return nil
+	}
+
+	if err := env.Open(path.Join(dirpath, DBStoreName), lmdb.NoSubdir, 0600); err != nil {
```

Comment:
@kacf IIRC you've used LMDB in cfengine, can you advise on the flags to use here and in places where `OpenRoot()` is called?